src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/CertificatePal.cs

```diff
@@ -40,20 +43,23 @@ public static ICertificatePal FromHandle(IntPtr handle)
         }
 
         /// <summary>
-        /// Returns the SafeCertContextHandle. Use this instead of FromHandle property when
+        /// Returns the SafeCertContextHandle. Use this instead of FromHandle() when
         /// creating another X509Certificate object based on this one to ensure the underlying
         /// cert context is not released at the wrong time.
         /// </summary>
         /// <param name="cert"></param>
         /// <returns></returns>
         public static ICertificatePal FromOtherCert(X509Certificate cert)
         {
-            CertificatePal newCert = (CertificatePal)FromHandle(cert.Handle);
-            newCert._certContextCloned = true;
+            CertificatePal originalPal = ((CertificatePal)cert.Pal);
+            originalPal._certContextCloned = true;
 
-            ((CertificatePal)cert.Pal)._certContextCloned = true;
+            // Use _certContext (instead of CertContext) to keep the original context handle from being
+            // finalized until all cert copies are no longer referenced.
+            CertificatePal newPal = new CertificatePal(originalPal._certContext, false);
```

Comment:
Since it's the last parameter you can just use the named syntax:

```C#
CertificatePal newPal = new CertificatePal(originalPal._certContext, deleteKeyContainer: false);
```